src/status_im/contexts/wallet/wallet_connect/events/sessions.cljs

```diff
@@ -109,25 +137,37 @@
               :on-success [:wallet-connect/fetch-persisted-sessions-success]
               :on-error   [:wallet-connect/fetch-persisted-sessions-fail]}]]]})))
 
+(rf/reg-event-fx :wallet-connect/persist-session-success
+ (fn [{:keys [db]} [session-info]]
+   (let [meta-data             (-> session-info
+                                   (js->clj :keywordize-keys true)
+                                   (get-in [:peer :metadata]))
+         redirect-url          (get-in meta-data [:redirect :native])
+         total-connected-dapps (-> db
+                                   :wallet-connect/sessions
+                                   count)]
+     (log/info "Wallet Connect session persisted")
+     {:db (assoc db
+                 :centralized-metrics/event-data
+                 {:dapp_name             (:name meta-data)
+                  :total_connected_dapps total-connected-dapps})
+      :fx [[:dispatch [:wallet-connect/fetch-persisted-sessions]
+            :dispatch [:wallet-connect/redirect-to-dapp redirect-url]]]})))
+
 (rf/reg-event-fx
  :wallet-connect/persist-session
  (fn [_ [session-info]]
-   (let [redirect-url (-> session-info
-                          (js->clj :keywordize-keys true)
-                          (data-store/get-dapp-redirect-url))]
-     {:fx [[:json-rpc/call
-            [{:method     "wallet_addWalletConnectSession"
-              :params     [(js/JSON.stringify session-info)]
-              :on-success (fn []
-                            (log/info "Wallet Connect session persisted")
-                            (rf/dispatch [:wallet-connect/fetch-persisted-sessions])
-                            (rf/dispatch [:wallet-connect/redirect-to-dapp redirect-url]))
-              :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]})))
+   {:fx [[:json-rpc/call
+          [{:method     "wallet_addWalletConnectSession"
+            :params     [(js/JSON.stringify session-info)]
+            :on-success (fn []
+                          (rf/dispatch [:wallet-connect/persist-session-success session-info]))
+            :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]}))
```

Comment:
We may be able to define on-success like: `:on-success [:wallet-connect/persist-session-success session-info]` (?)